lib/spack/spack/binary_distribution.py

```diff
@@ -935,15 +935,22 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
     spackfile_path = os.path.join(
         cache_prefix, tarball_path_name(spec, '.spack'))
 
-    remote_spackfile_path = url_util.join(
+    outdir = mirror.push_url
```

Comment:
Before this change, the `outdir` passed into this method was created like:

```
outdir = url_util.format(mirror.push_url)
```

I'm not sure it's critical, but to me it seems there's no reason to change it arbitrarily.  Let me know if there's something I'm missing.

Reply:
You're right, I'll fix this.